homeassistant/components/bryant_evolution/climate.py

```diff
@@ -0,0 +1,293 @@
+                        # then we must be cooling.
+                        return HVACAction.COOLING
+                    return HVACAction.HEATING
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="failed_to_parse_hvac_mode",
+            translation_placeholders={
+                "mode_and_active": mode_and_active,
+                "current_temperature": str(self.current_temperature),
+                "target_temperature_low": str(self.target_temperature_low),
+            },
+        )
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == HVACMode.HEAT_COOL:
+            hvac_mode = HVACMode.AUTO
+        if not await self._client.set_hvac_mode(hvac_mode):
+            raise HomeAssistantError(
+                translation_domain=DOMAIN, translation_key="failed_to_set_hvac_mode"
+            )
+        self._attr_hvac_mode = hvac_mode
+        self._async_write_ha_state()
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        if kwargs.get("target_temp_high"):
+            temp = int(kwargs["target_temp_high"])
+            if not await self._client.set_cooling_setpoint(temp):
+                raise HomeAssistantError(
+                    translation_domain=DOMAIN, translation_key="failed_to_set_clsp"
+                )
+            self._attr_target_temperature_high = temp
+            self._async_write_ha_state()
+
+        if kwargs.get("target_temp_low"):
+            temp = int(kwargs["target_temp_low"])
+            if not await self._client.set_heating_setpoint(temp):
+                raise HomeAssistantError(
+                    translation_domain=DOMAIN, translation_key="failed_to_set_htsp"
+                )
+            self._attr_target_temperature_low = temp
+            self._async_write_ha_state()
+
+        if kwargs.get("temperature"):
+            temp = int(kwargs["temperature"])
+            fn = (
+                self._client.set_heating_setpoint
+                if self.hvac_mode == HVACMode.HEAT
+                else self._client.set_cooling_setpoint
+            )
+            if not await fn(temp):
+                raise HomeAssistantError(
+                    translation_domain=DOMAIN, translation_key="failed_to_set_temp"
+                )
+            self._attr_target_temperature = temp
+            self._async_write_ha_state()
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        if not await self._client.set_fan_mode(fan_mode):
+            raise HomeAssistantError(
+                translation_domain=DOMAIN, translation_key="failed_to_set_fan_mode"
+            )
+        self._attr_fan_mode = fan_mode.lower()
+        self.async_write_ha_state()
+
+
+class BryantEvolutionSystem(Entity):
```

Comment:
Wait what entities does this have? Shouldn't this be a device like the SAM as well?

Reply:
Sorry, I should have left a comment -- there are a bunch of entities I can add in a future PR (potentially mostly default-disabled). Things like filter life remaining, humidifier status (if installed), vacation mode...

Reply:
removing for now as discussed